### PR TITLE
Assistant: UI Metric Updates

### PR DESCRIPTION
**Pull request by mc-wright**
- Added various UI metrics and other updates to the Assistant and AI Metrics pages. [#358](https://github.com/Security-Onion-Solutions/sos/issues/358)
  - Total credits for a session now appears next to context length.
  - Duration column for sessions on the AI Metrics page; `1d 1h 1m` format.
  - Credits per minute column for sessions on the AI Metrics page.
  - Current model is now shown above context limit.
  - "Credits Remaining" at the top right now has a loading spinner when loading the page.
  - Reworded Assistant error message.
- Modified toggle_detections tool to use bool instead of string for "enable" input.
- Fixed issue where AI Metrics was erroring for sessions with outdated formatting or other edge cases.
- Added test coverage for everything above.

### CODE REVIEW

File: server/modules/elastic/elasticassistantstore.go [hunk 1]
```diff
@@ -697,6 +702,134 @@ func (store *ElasticAssistantstore) populateSessionUsage(ctx context.Context, se
...
+		logger.WithError(err).Error("Failed to unmarshal MSearch response")
+		return err
+	}
+
+	// Parse responses and populate session usage
+	if responses, ok := response["responses"].([]any); ok {
+		for i, respObj := range responses {
+			if i >= len(sessions) {
+				break
+			}
```
Comment (coreyogburn) [inline server/modules/elastic/elasticassistantstore.go:796]: Does this happen a lot? It should have exactly 1 response per sent query.

File: html/js/routes/aimetrics.js [hunk 2]
```diff
@@ -500,5 +508,24 @@ routes.push({ path: '/aimetrics/:userId?/:sessionId?', name: 'aimetrics', compon
...
+      const minutes = Math.floor(seconds / 60);
+
+      let result = `${days}d ${hours}h ${minutes}m`;
+      return result;
+    },
+    getCPM(durationMs, totalCredits) {
+      let minutes = durationMs / 60000;
+      let rawCPM = totalCredits / minutes;
+      return Math.round(rawCPM);
+    }
```
Comment (coreyogburn) [inline html/js/routes/aimetrics.js:516]: A duration of 0 returns infinity. Maybe durations less than a minute should be calculated as a minute?

File: server/modules/elastic/elasticassistantstore.go (hunk 1 above)
Comment (mc-wright) [reply server/modules/elastic/elasticassistantstore.go:796]: It should never happen. It's just a safety check in case we add more to this function in the future, and it doesn't add any meaningful complexity to the runtime.

File: html/js/routes/aimetrics.js (hunk 2 above)
Comment (mc-wright) [reply html/js/routes/aimetrics.js:516]: fixed

File: html/js/routes/aimetrics.js [hunk 3]
```diff
@@ -500,5 +508,25 @@ routes.push({ path: '/aimetrics/:userId?/:sessionId?', name: 'aimetrics', compon
       let percentage = (credits / total) * 100;
       return `${percentage.toFixed(1)}%`;
     },
+    formatDHM(durationMs) {
```
Comment (jertel) [inline html/js/routes/aimetrics.js]: Move this to app.js next to the other date formatters. Call it formatLongDuration(). Ideally, it would show only "d" when > 24h, only "h" when > 60m, etc.

Ex: 1.5d
Ex: 2.3h
Ex: 45m
Ex: 1m

Comment (mc-wright) [reply html/js/routes/aimetrics.js]: Done

**Review by jertel — APPROVED** [review]
